server/s3_copy_object_action.cc

```diff
@@ -292,13 +303,76 @@ void S3CopyObjectAction::read_data_block_success() {
     return;
   }
   if (copy_failed) {
+    assert(!write_in_progress);
+
     send_response_to_s3_client();
     return;
   }
+  assert(data_blocks_read.empty());
+  data_blocks_read = motr_reader->extract_blocks_read();
+
+  size_t bytes_in_chunk_count = 0;
+
+  if (data_blocks_read.empty()) {
+    s3_log(S3_LOG_ERROR, stripped_request_id, "Motr reader returned no data");
+
+    s3_put_action_state = S3PutObjectActionState::writeFailed;
+    copy_failed = true;
+    set_s3_error("InternalError");
+
+    if (!write_in_progress) {
+      send_response_to_s3_client();
+    }
+    return;
+  }
+  // Calculating actial size of data that has just been read
+
+  for (size_t i = 0, n = data_blocks_read.size() - 1; i < n; ++i) {
+    const auto motr_block_size = data_blocks_read[i].second;
+    assert(motr_block_size == motr_unit_size);
+
+    // We can use multiplication, but something may change in the future
+    bytes_in_chunk_count += motr_block_size;
+  }
+  if (bytes_in_chunk_count >= bytes_left_to_read) {
+    s3_log(S3_LOG_ERROR, stripped_request_id, "Too many data has been read");
```

Comment:
Too **much**

Reply:
I am assuming this situation should not have happened. And we are in this if block because something is not good, and instead of discarding the extra read bytes, its good to set error here.

Is this understanding correct ?

Reply:
I thought we should detect such issuers. We can face with bugs in S3MotrReader class or Motr code.

Reply:
yes, I totally agree... But I wanted to understand is this situation ("too much read") is because of some corruption/problem in the system/stack or can happen commonly ?